src/data/list/basic.lean

```diff
@@ -2746,6 +2751,11 @@ nat.le.intro (head_add_tail_sum L)
 lemma tail_sum (L : list ℕ) : L.tail.sum = L.sum - L.head :=
 by rw [← head_add_tail_sum L, add_comm, add_tsub_cancel_right]
 
+-- The product of a list of positive natural numbers is positive
+lemma pos_list_nat_prod_pos (L : list ℕ) (h: ∀ n ∈ L, 0 < n) : 0 < L.prod :=
```

Comment:
This should be renamed `prod_pos` and `nat` should be generalized to any `nontrivial` `ordered_semiring` (prove it by induction on the list and use `mul_pos`).

Reply:
Thanks, I've made this generalisation